src/cpu/platform.cpp

```diff
@@ -195,7 +195,23 @@ unsigned get_per_core_cache_size(int level) {
 
 unsigned get_num_cores() {
 #if DNNL_X64
-    return x64::cpu().getNumCores(Xbyak::util::CoreLevel);
+    int num_cores = x64::cpu().getNumCores(Xbyak::util::CoreLevel);
+    // Sometimes getNumCores() can't detect number of cores and returns
+    // 0 (e.g. in AMD Systems), so we need to read the system info
+    if (num_cores > 0) { return num_cores; }
+#ifdef _WIN32
```

Comment:
hi Igor, `getNumCores()` works at different topology levels. In our example, we are looking at the core-level. The logic I added will only work for the core-level. So, won't it make sense to make the change here other than `getNumCores()`?

Reply:
As you said `getNumCores()` works at multiple levels. One of them is `CoreLevel`, so I thought that this code can go in that section:
https://github.com/herumi/xbyak/blob/a1ac3750f9a639b5a6c6d6c7da4259b8d6790989/xbyak/xbyak_util.h#L273 

@herumi Do you think number of cores detection for AMD can be integrated into XByak directly?

Reply:
If I can know how to detect the number of cores of AMD, then I'll implement (or merge) it.

Reply:
sys._info.dwNumberOfProcessors and cpu_count show a number of logical CPUs but `getNumCores(Xbyak::util::CoreLevel)` shows a number of physical CPUs per socket. Is that what you intended?
And Xbyak::Cpu does not consider heterogeneous CPUs like an Alder Lake now.

Reply:
I second the idea to put the logic of counting physical cores in Xbyak directly, and relying on CPUID information.

The current code to count the number of physical cores in Xbyak ([here](https://github.com/herumi/xbyak/blob/a1ac3750f9a639b5a6c6d6c7da4259b8d6790989/xbyak/xbyak_util.h#L176)) and detect CPU/cache hierarchy ([here](https://github.com/herumi/xbyak/blob/a1ac3750f9a639b5a6c6d6c7da4259b8d6790989/xbyak/xbyak_util.h#L214)), currently skip non-intel CPUs as AMD CPUs use a different CPUID logic for core/cache enumeration.

Could you please add the CPUID enumeration there directly, so that all implementation will be able to rely on this information directly?

Reply:
> it seems it has already been done in that fork.

I merged it into [a dev branch](https://github.com/herumi/xbyak/commit/edce727092cee5257bfee100f6ae2e6894fdcdf9). How about it?

Reply:
I've released [Xbyak 6.69](https://github.com/herumi/xbyak).

Reply:
Thanks @herumi !

Reply:
@bviyer Xbyak was updated to v6.69.1 ( 9197c015e60324a5c4c14b818168abf0be8cd186 )
Could you try agin with the latest version? Now Xbyak should correctly detect core & cache configuration on AMD.